spark/src/test/scala/org/apache/hadoop/fs/versioned/VersionedFileSystemSpec.scala

```diff
@@ -0,0 +1,73 @@
+package org.apache.hadoop.fs.versioned
+
+import com.gu.tableversions.core.Partition
+import com.gu.tableversions.core.Partition.{ColumnValue, PartitionColumn}
+import com.gu.tableversions.metastore.VersionPaths
+import com.gu.tableversions.spark.SparkHiveSuite
+import org.apache.spark.sql.SaveMode
+import org.scalatest.{FlatSpec, Matchers}
+
+class VersionedFileSystemSpec
+    extends FlatSpec
+    with Matchers
+    with SparkHiveSuite
+    with S3TestSuite
+    with S3SparkTestSuite {
```

Comment:
I think we should have a think about how to unit test the `VersionedFilesystem` class. The test below that writes via Spark is useful but I think some direct unit tests for the class itself would be valuable too.